env0/resource_environment.go

```diff
@@ -87,7 +87,7 @@ func resourceEnvironment() *schema.Resource {
 			},
 			"auto_deploy_by_custom_glob": {
 				Type:        schema.TypeString,
-				Description: "redeploy on file filter pattern",
+				Description: "redeploy on file filter pattern. If used then 'auto_deploy_on_path_changes_only' must be configured to true. If used then 'deploy_on_push' or 'run_plan_on_pull_requests' must be configured to true.",
```

Comment:
I'm thinking maybe we should add the following validations:

Inside `auto_deploy_by_custom_glob`:
- add RequiredWith `auto_deploy_on_path_changes_only`
- add AtLeastOneOf `deploy_on_push , run_plan_on_pull_requests`

WDYT?

Reply:
Won't work see the following:

```
	// AtLeastOneOf is a set of attribute paths, including this attribute,
	// in which at least one of the attributes must be configured. This
	// implements the validation logic declaratively within the schema and can
	// trigger earlier in Terraform operations, rather than using create or
	// update logic which only triggers during apply.
```

I will add the `RequiredWith`.